src/frame/staking.rs

```diff
@@ -0,0 +1,315 @@
+    pub stash: T::AccountId,
+}
+
+/// The map from (wannabe) validator stash key to the preferences of that validator.
+#[derive(Encode, Copy, Clone, Debug, Hash, PartialEq, Eq, Ord, PartialOrd, Store)]
+pub struct ValidatorsStore<T: Staking> {
+    #[store(returns = ValidatorPrefs)]
+    /// Tٗhe stash account
+    pub stash: T::AccountId,
+}
+
+/// The map from nominator stash key to the set of stash keys of all validators to nominate.
+#[derive(Encode, Copy, Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Store)]
+pub struct NominatorsStore<T: Staking> {
+    #[store(returns = Option<Nominations<T::AccountId>>)]
+    /// Tٗhe stash account
+    pub stash: T::AccountId,
+}
+
+/// The current era index.
+///
+/// This is the latest planned era, depending on how the Session pallet queues the validator
+/// set, it might be active or not.
+#[derive(Encode, Copy, Clone, Debug, Store)]
+pub struct CurrentEraStore<T: Staking> {
+    #[store(returns = Option<EraIndex>)]
+    /// Marker for the runtime
+    pub _runtime: PhantomData<T>,
+}
+
+/// Reward points of an era. Used to split era total payout between validators.
+///
+/// This points will be used to reward validators and their respective nominators.
+#[derive(PartialEq, Encode, Decode, Default, Debug)]
+pub struct EraRewardPoints<AccountId: Ord> {
+    /// Total number of points. Equals the sum of reward points for each validator.
+    pub total: RewardPoint,
+    /// The reward points earned by a given validator.
+    pub individual: BTreeMap<AccountId, RewardPoint>,
+}
+
+/// Clipped Exposure of validator at era.
+///
+/// This is similar to [`ErasStakers`] but number of nominators exposed is reduced to the
+/// `T::MaxNominatorRewardedPerValidator` biggest stakers.
+/// (Note: the field `total` and `own` of the exposure remains unchanged).
+/// This is used to limit the i/o cost for the nominator payout.
+///
+/// This is keyed fist by the era index to allow bulk deletion and then the stash account.
+///
+/// Is it removed after `HISTORY_DEPTH` eras.
+/// If stakers hasn't been set or has been removed then empty exposure is returned.
+#[derive(Encode, Copy, Clone, Debug, Store)]
+pub struct ErasStakersClippedStore<T: Staking> {
+    #[store(returns = Exposure<T::AccountId, T::Balance>)]
+    /// Era index
+    pub era: EraIndex,
+    /// Stash account of the validator
+    pub validator_stash: T::AccountId,
+}
+
+/// The active era information, it holds index and start.
+///
+/// The active era is the era currently rewarded.
+/// Validator set of this era must be equal to `SessionInterface::validators`.
+#[derive(Encode, Copy, Clone, Debug, Hash, PartialEq, Eq, Ord, PartialOrd, Store)]
+pub struct ActiveEraStore<T: Staking> {
+    #[store(returns = Option<ActiveEraInfo>)]
+    /// Marker for the runtime
+    pub _runtime: PhantomData<T>,
+}
+
+/// Declare the desire to validate for the origin controller.
+///
+/// Effects will be felt at the beginning of the next era.
```

Comment:
```suggestion
/// Effective from the beginning of the next era.
```
or perhaps "If successful, effective from the beginning of the next era."?